src/textual/widgets/_option_list.py

```diff
@@ -647,6 +655,51 @@ def remove_option_at_index(self, index: int) -> Self:
             ) from None
         return self
 
+    def _replace_option_prompt(self, index: int, prompt: RenderableType) -> None:
+        """Replace the prompt of an option in the list.
+
+        Args:
+            index: The index of the option to replace the prompt of.
+            prompt: The new prompt for the option.
+
+        Raises:
+            IndexError: If there is no option of the given index.
+        """
+        self.get_option_at_index(index).set_prompt(prompt)
+        self._refresh_content_tracking(force=True)
+        self.refresh()
+
+    def replace_option_prompt(self, option_id: str, prompt: RenderableType) -> Self:
+        """Replace the prompt of the option with the given ID.
+
+        Args:
+            option_id: The ID of the option to replace the prompt of.
+            prompt: The new prompt for the option.
+
+        Returns:
+            The `OptionList` instance.
```

Comment:
Can we also document what gets raised if the option doesn't exist?

Reply:
Is it necessary if it's already documented in the `get_option_index()` method?

Reply:
I would say "yes" because users will see the documentation for `replace_option_prompt` and they won't look at how it is implemented to figure out what may get raised.
This will also make sure this method is more aligned with all others, because all other methods say they will raise the error. It's just that we raise it a bit later.

Reply:
Ok, you are right. I added `Raises` section.